Test fails when there is a `tests*` folder containing no testrunner-meaning tests under site-packages

I got some mystical errors when running the testsuite of zope.testrunner, and finally found the problem to be a `/usr/lib/python3.5/site-packages/tests` folder (installed by betamax, and I've opened an issue there for this). Errors are gone immediately after I moved that folder away. (Still doesn't work if renaming it to something like `tests2`, though)

Errors:

```
running test
running egg_info
writing entry points to src/zope.testrunner.egg-info/entry_points.txt
writing requirements to src/zope.testrunner.egg-info/requires.txt
writing top-level names to src/zope.testrunner.egg-info/top_level.txt
writing namespace_packages to src/zope.testrunner.egg-info/namespace_packages.txt
writing src/zope.testrunner.egg-info/PKG-INFO
writing dependency_links to src/zope.testrunner.egg-info/dependency_links.txt
package init file 'src/zope/testrunner/tests/testrunner-ex/__init__.py' not found (or not a regular file)
reading manifest template 'MANIFEST.in'
writing manifest file 'src/zope.testrunner.egg-info/SOURCES.txt'
running build_ext
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.


Failure in test /home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-layer-setup.test
Failed doctest test for testrunner-debugging-layer-setup.test
  File "/home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-layer-setup.test", line 0

----------------------------------------------------------------------
File "/home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-layer-setup.test", line 51, in testrunner-debugging-layer-setup.test
Failed example:
    try:
        zope.testrunner.run_internal(['--path', dir, '-D'])
    except zope.testrunner.interfaces.EndRun: print("unexpected EndRun!")
    finally: sys.stdin = real_stdin
    # doctest: +ELLIPSIS
Expected:
    Running tests.Layer tests:
      Set up tests.Layer ...ValueError
    <BLANKLINE>
    > ...tests.py(NNN)setUp()
    -> raise ValueError
    (Pdb) p x
    1
    (Pdb) c
    False
Got:
    TypeError: Module tests does not define any tests
    <BLANKLINE>
    > /home/felix/projects/other/zope.testrunner/src/zope/testrunner/find.py(NNN)find_suites()
    -> % module_name)
    (Pdb) p x
    *** NameError: name 'x' is not defined
    (Pdb) c
    unexpected EndRun!



Failure in test /home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-import-failure.test
Failed doctest test for testrunner-debugging-import-failure.test
  File "/home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-import-failure.test", line 0

----------------------------------------------------------------------
File "/home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-import-failure.test", line 36, in testrunner-debugging-import-failure.test
Failed example:
    try:
        zope.testrunner.run_internal(['--path', dir, '-D'])
    except zope.testrunner.interfaces.EndRun: print('EndRun raised')
    finally: sys.stdin = real_stdin
    # doctest: +ELLIPSIS +REPORT_NDIFF
Differences (ndiff with -expected +actual):
    + TypeError: Module tests does not define any tests
    + <BLANKLINE>
    + > /home/felix/projects/other/zope.testrunner/src/zope/testrunner/find.py(NNN)find_suites()
    + -> % module_name)
    -   File ".../TESTS-DIR/tests.py", line 2
    -     impot doctest
    -                 ^
    - SyntaxError: invalid syntax
    - > ...find.py(NNN)import_name()
    - -> __import__(name)
      (Pdb) c
      EndRun raised



Failure in test /home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-nonprintable-exc.test
Failed doctest test for testrunner-debugging-nonprintable-exc.test
  File "/home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-nonprintable-exc.test", line 0

----------------------------------------------------------------------
File "/home/felix/projects/other/zope.testrunner/src/zope/testrunner/tests/testrunner-debugging-nonprintable-exc.test", line 41, in testrunner-debugging-nonprintable-exc.test
Failed example:
    try:
        zope.testrunner.run_internal(['--path', dir, '-D'])
    except zope.testrunner.interfaces.EndRun: print('EndRun raised')
    finally: sys.stdin = real_stdin
    # doctest: +ELLIPSIS +REPORT_NDIFF
Differences (ndiff with -expected +actual):
    - Running zope.testrunner.layer.UnitTests tests:
    - ...
    - AssertionError: ...'a' != ...'...'
    - ...
    + TypeError: Module tests does not define any tests
    + <BLANKLINE>
    + > /home/felix/projects/other/zope.testrunner/src/zope/testrunner/find.py(NNN)find_suites()
    + -> % module_name)
      (Pdb) c
    + EndRun raised
    - Tearing down left over layers:
    - ...
    - False

  Ran 53 tests with 3 failures, 0 errors, 0 skipped in 16.176 seconds.
Tearing down left over layers:
  Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.
```

--- a/src/zope/testrunner/find.py
+++ b/src/zope/testrunner/find.py
@@ -472,9 +472,9 @@ class Find(zope.testrunner.feature.Feature):
 
     def global_setup(self):
         # Add directories to the path
-        for path in self.runner.options.path:
+        for path in reversed(self.runner.options.path):
             if path not in sys.path:
-                sys.path.append(path)
+                sys.path.insert(0, path)
 
         tests = find_tests(self.runner.options, self.runner.found_suites)
         self.import_errors = tests.pop(None, None)

